### PR TITLE
Add lastUsedAt field in OAuthAuthorization

### PR DESCRIPTION
**Pull request by la-costa**
Implements the back end functionality for show last usage date for connected application at front end.

### CODE REVIEW

File: server/graphql/v2/object/OAuthAuthorization.js [hunk 1]
```diff
@@ -38,6 +39,19 @@ export const OAuthAuthorization = new GraphQLObjectType({
       type: new GraphQLNonNull(GraphQLDateTime),
       description: 'The time of expiration',
     },
+    lastUsedAt: {
+      type: GraphQLDateTime,
+      description: 'The last time of token was used',
+      async resolve(authorization) {
+        const activity = await models.Activity.findOne({
```
Comment (Betree) [inline server/graphql/v2/object/OAuthAuthorization.js:46]: We should add a permission check here; only an admin of the token should be able to know its last usage date

File: server/graphql/v2/object/OAuthAuthorization.js [hunk 2]
```diff
@@ -38,6 +39,19 @@ export const OAuthAuthorization = new GraphQLObjectType({
...
+          where: {
```
Comment (Betree) [inline server/graphql/v2/object/OAuthAuthorization.js:47]: Small perf improvements: add a `SELECT` on the query to get only the field we're interested in

```suggestion
          attributes: ['createdAt'],
          where: {
```

**Review by Betree — CHANGES_REQUESTED** [review]
Looking good!

To get rid of the ESLint error, we should run `npm run graphql:update` while the API is running.

File: server/graphql/v2/object/OAuthAuthorization.js (hunk 1 above)
Comment (la-costa) [reply server/graphql/v2/object/OAuthAuthorization.js]: I added the permission check, can you check if the implementation is correct?

I also already updated GraphQL

File: test/server/graphql/v2/query/ApplicationQuery.test.ts [hunk 3]
```diff
@@ -16,6 +16,7 @@ const applicationQuery = gqlV2/* GraphQL */ `
       redirectUri
       clientId
       clientSecret
+      lastUsedAt
```
Comment (Betree) [inline test/server/graphql/v2/query/ApplicationQuery.test.ts:19]: `lastUsedAt` lives in `OAuthAuthorization`, not `Application` (this is the reason why tests are failing)

Comment (la-costa) [reply test/server/graphql/v2/query/ApplicationQuery.test.ts]: Thanks for the note, I already fixed it.

However, one of the tests is still failing and the reason is that the test data is not correctly instantiated. I tried to create the `OAuthAuthorization`, but I was not successful. Can you help me at this point?

Comment (Betree) [reply test/server/graphql/v2/query/ApplicationQuery.test.ts]: Thanks, I'll take it from here

**Comment by Betree** [comment]
Issue: https://github.com/opencollective/opencollective/issues/5650

**Pull request merged**